src/languages/swift.js

```diff
@@ -283,6 +288,143 @@ export default function(hljs) {
   };
   TYPE.contains.push(GENERIC_ARGUMENTS);
 
+  // https://docs.swift.org/swift-book/ReferenceManual/Expressions.html#ID552
+  // Prevents element names from being highlighted as keywords.
+  const TUPLE_ELEMENT_NAME = {
+    begin: lookahead(concat(Swift.identifier, /\s*:/)),
+    end: lookahead(/:/),
+    keywords: "_",
+    relevance: 0
+  };
+  // Matches tuples as well as the parameter list of a function type.
+  const TUPLE = {
+    begin: /\(/,
+    end: /\)/,
+    relevance: 0,
+    keywords: KEYWORDS,
+    contains: [
+      'self',
+      TUPLE_ELEMENT_NAME,
+      ...COMMENTS,
+      ...KEYWORD_MODES,
+      ...BUILT_INS,
+      ...OPERATORS,
+      NUMBER,
+      STRING,
+      ...IDENTIFIERS,
+      ...ATTRIBUTES,
+      TYPE
+    ]
+  };
+
+  // https://docs.swift.org/swift-book/ReferenceManual/Declarations.html#ID362
+  // Matches both the keyword func and the function title.
+  // Grouping these lets us differentiate between the operator function <
+  // and the start of the generic parameter clause (also <).
+  const FUNCTION_TITLE = {
+    begin: lookahead(/\bfunc\b/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\bfunc\b/
+      },
+      {
+        begin: /\s+/,
+        relevance: 0
+      },
+      {
+        className: 'title',
+        begin: either(QUOTED_IDENTIFIER.begin, Swift.identifier, Swift.operator),
+        // Required to make sure the opening < of the generic parameter clause
+        // isn't parsed as a second title.
+        endsParent: true,
+        relevance: 0
+      }
+    ]
+  };
+  const GENERIC_PARAMETERS = {
+    begin: /\s*</,
+    end: />/,
+    contains: [
+      ...COMMENTS,
+      TYPE
+    ]
+  };
+  const FUNCTION_PARAMETER_NAME = {
+    begin: either(
+      lookahead(concat(Swift.identifier, /\s*:/)),
+      lookahead(concat(Swift.identifier, /\s+/, Swift.identifier, /\s*:/))
+    ),
+    end: lookahead(/:/),
+    relevance: 0,
+    contains: [
+      {
+        className: 'keyword',
+        begin: /\b_\b/
+      },
+      {
+        className: 'params',
+        begin: Swift.identifier
+      }
+    ]
+  };
+  const FUNCTION_PARAMETERS = {
+    begin: /\s*\(/,
+    end: /\)/,
+    keywords: KEYWORDS,
+    contains: [
+      FUNCTION_PARAMETER_NAME,
+      ...COMMENTS,
+      ...KEYWORD_MODES,
+      ...OPERATORS,
+      NUMBER,
+      STRING,
+      ...ATTRIBUTES,
+      TYPE,
+      TUPLE
+    ],
+    illegal: /["']/
+  };
+  const FUNCTION = {
+    className: 'function',
+    begin: lookahead(/\bfunc\b/),
+    contains: [
+      FUNCTION_TITLE,
+      GENERIC_PARAMETERS,
+      FUNCTION_PARAMETERS
+    ],
```

Comment:
```suggestion
    contains: [
      FUNCTION_TITLE,
      { begin: /\s*/ }, // allow whitespace
      GENERIC_PARAMETERS,
      FUNCTION_PARAMETERS
    ],
```

Or you could make a constant of course, but single place.  Done. Much simpler.

Reply:
There are also ways to force end to hold open but then you'd need a child rule with `endsParent`... either of those those options would be preferable to appending `\s*` multiple places.  I think it's very explicit (and hence easier to understand) listing it right inside the contains though.

If I did it implicitly (holding end open) I'd want to add a comment explaining it was for whitespace anyways, so seems simpler to just have the rule right there.

Reply:
Fixed. See explanation above.